diesel/src/query_builder/upsert/on_conflict_actions.rs

```diff
@@ -18,11 +18,32 @@ where
     }
 }
 
-impl<DB, T> QueryFragment<DB, T> for DoNothing
+impl<DB> QueryFragment<DB, sql_dialect::on_conflict_clause::PgLikeOnConflictClause> for DoNothing
```

Comment:
This breaks support for `DO NOTHING` clauses on sqlite. We should likely restrict `T` there to some new trait that's only implemented for `PgLikeOnConflictClause` and `SqliteOnConflictClause`.

Reply:
Yeah... that's I think what I'm trying to do; see my response to [your earlier comment](https://github.com/diesel-rs/diesel/pull/3177/files#r875224280)

Reply:
So what we want here is something like:
```rust 
impl<DB, S> QueryFragment<DB, S> for DoNothing 
where S: PostgresLikeDoNothing
{…}
```
with a newly introduced `PostgresLikeDoNothing` (feel free to choose a better name :wink:) trait, that's implemented for `PgLikeOnConflictClause` and `SqliteLikeOnConflictClause`.

This allows to provide another impl for the mysql case, as long as the mysql `OnConflict` marker type does not implement `PostgresLikeDoNothing`, while the other two backends can share the corresponding implementation.

Reply:
Ugh I couldn't figure out how to do this and was flailing a little too much 😅 . Instead I'm opting to not support `DO NOTHING` on MySQL for now -- I figure any users can workaround the issue by performing an effective noop update if they need to